src/nanotron/config/config.py

```diff
@@ -210,6 +221,7 @@ class TokensArgs:
     batch_accumulation_per_replica: int
 
     val_check_interval: Optional[int] = -1
+    val_steps: Optional[int] = 10
```

Comment:
This is another detail I wanted to discuss. Currently, the `len(train_dataset)` is calculated as the `number of training steps * global batch size`, while for evaluation/test, what you need is to specify how often you want it to be conducted and how many steps you want to perform. From here, you can determine the total number of samples you need to create each dataset.

An example will make this clearer: Let's start with a preprocessed dataset that has 100 documents, and with a split of `8,1,1`, 80 documents will go to train, 10 to valid, and 10 to test. With these 80 documents for training, let's say you extract 50 training samples. If you have a global batch size of 8 and 10 training steps, you will need to generate 80 samples, meaning you will train for 1.4 epochs. 

However, for validation, if you have an `val_check_interval` of 2, during 10 training steps, you will conduct the evaluation 6 times. In these 6 evaluations, you will perform `val_steps`, which if it's 10, you will need your valid dataset to have `6 * 10 * global batch size` samples. Considering that perhaps from the 10 documents we used in the example, you only extract 5 samples and the global batch size is 8, in each of the 6 validation checks, you consume 80 samples, which means you will consume those 5 samples 12 times.

I don't see the point in this method, as I believe it would be better for both validation and test to consume the entire dataset only once. I will modify it.

Reply:
I have updated the logic behind the number of samples for each split. Finally, in `NanosetConfigs`, we introduce the quantity of `train_samples` (computed as before, `train_steps * global_batch_size`. Previously, we introduced `split_num_samples` as a list with the quantity of samples per split calculated as I mentioned in my previous message) and from here, after initialization we will create`self.split_num_samples = [train_split_samples, None, None]`. These `None` values will indicate to the Nanoset constructor that its samples will only be consumed once, that is, we will not concatenate the indices as we do for the train split. This way, the `DataLoader`will serve all samples only once.

For the case of the `BlendedNanoset`, based on the value of `train_split_samples` and with the `weights` of each dataset, we will compute the number of samples to extract from each dataset in the blend. In this case, it should be noted that it is possible that one dataset from the blend is consumed 5 times while another larger one is consumed only once. For the valid and test split, we will consume all the samples reserved for these splits only once.

I have added all to the docs.